### PR TITLE
CCIP-2882 Styling fixes and post review items 

### PR DESCRIPTION
**Pull request by mateusz-sekara**

### CODE REVIEW

File: execute/tokendata/observer.go [hunk 1]
```diff
@@ -46,23 +46,21 @@ type compositeTokenDataObserver struct {
...
 		// TODO consider if we can get rid of this switch stmt by moving the logic to the config
 		// e.g. observers[i] := config.CreateTokenDataObserver()
 		switch {
 		case c.USDCCCTPObserverConfig != nil:
-			observer, err1 := createUSDCTokenObserver(lggr, destChainSelector, *c.USDCCCTPObserverConfig, readers)
-			if err1 != nil {
-				return nil, err1
+			observer, err := createUSDCTokenObserver(lggr, destChainSelector, *c.USDCCCTPObserverConfig, readers)
+			if err != nil {
+				return nil, err
```
Comment (dimkouv) [inline execute/tokendata/observer.go:63]: wrap err?

File: execute/tokendata/observer.go [hunk 2]
```diff
@@ -86,7 +84,7 @@ func createUSDCTokenObserver(
 		return nil, err
 	}
 
-	client, err := usdc.NewAttestationClient(cctpConfig)
+	client, err := usdc.NewSequentialAttestationClient(cctpConfig)
 	if err != nil {
 		return nil, err
```
Comment (dimkouv) [inline execute/tokendata/observer.go:89]: wrap err

File: execute/tokendata/usdc/attestation.go [hunk 3]
```diff
@@ -59,15 +62,14 @@ type sequentialAttestationClient struct {
...
-func NewAttestationClient(config pluginconfig.USDCCCTPObserverConfig) (*sequentialAttestationClient, error) {
+func NewSequentialAttestationClient(config pluginconfig.USDCCCTPObserverConfig) (AttestationClient, error) {
 	client, err := NewHTTPClient(
 		config.AttestationAPI,
 		config.AttestationAPIInterval.Duration(),
 		config.AttestationAPITimeout.Duration(),
 	)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to create HTTP client: %w", err)
```
Comment (dimkouv) [inline execute/tokendata/usdc/attestation.go:72]: general tip, no need to say `failed to...` in an error. "create http client: %w" is fine, it's an error anyways

File: execute/tokendata/usdc/http.go [hunk 4]
```diff
@@ -98,83 +98,89 @@ func (r httpResponse) validate() error {
...
 	var response httpResponse
 	if err := json.NewDecoder(res.Body).Decode(&response); err != nil {
-		return nil, res.StatusCode, fmt.Errorf("failed to decode json: %w", err)
+		return nil, fmt.Errorf("failed to decode json: %w", err)
 	}
 
-	if err1 := response.validate(); err1 != nil {
-		return nil, res.StatusCode, err1
+	if err := response.validate(); err != nil {
+		return nil, err
```
Comment (dimkouv) [inline execute/tokendata/usdc/http.go:180]: wrap err

File: execute/tokendata/usdc/http.go [hunk 5]
```diff
@@ -196,3 +202,5 @@ func (h *httpClient) inCoolDownPeriod() bool {
 	defer h.coolDownMu.RUnlock()
 	return time.Now().Before(h.coolDownUntil)
 }
+
+var _ HTTPClient = (*httpClient)(nil)
```
Comment (dimkouv) [inline execute/tokendata/usdc/http.go:206]: compliance check not required, constructor returns an interface instance, it should fail there

**Review by dimkouv — APPROVED** [review]
Approving, with some comments.

File: execute/tokendata/usdc/http.go (hunk 4 above)
Comment (mateusz-sekara) [reply execute/tokendata/usdc/http.go:180]: validate has already nicely formatter error, no need to wrap

**Pull request merged**